pkg/permissions/set.go

```diff
@@ -137,3 +138,25 @@ func (ps *Set) IsSubSetOf(parent Set) bool {
 
 	return true
 }
+
+// HasSameRules returns true if the two sets have exactly the same rules.
+func (ps Set) HasSameRules(other Set) bool {
```

Comment:
In that case, you should also check that `otherRule.Verbs` and `rule.Verbs` are really equal below